modules/core/04-channel/keeper/keeper_test.go

```diff
@@ -81,6 +82,73 @@ func (suite *KeeperTestSuite) TestGetAppVersion() {
 	suite.Require().Equal(ibcmock.Version, channelVersion)
 }
 
+// TestGetAllChannelsWithPortPrefix verifies ports are filtered correctly using a port prefix.
+func (suite *KeeperTestSuite) TestGetAllChannelsWithPortPrefix() {
+	tests := []struct {
+		name             string
+		prefix           string
+		allChannels      []types.IdentifiedChannel
+		expectedChannels []types.IdentifiedChannel
+	}{
+		{
+			name:   "transfer channel is retrieved with prefix",
+			prefix: "tra",
+			allChannels: []types.IdentifiedChannel{
+				types.NewIdentifiedChannel(transfertypes.TypeMsgTransfer, ibctesting.FirstChannelID, types.Channel{}),
+				types.NewIdentifiedChannel("different-portid", "channel-1", types.Channel{}),
+			},
+			expectedChannels: []types.IdentifiedChannel{types.NewIdentifiedChannel(transfertypes.TypeMsgTransfer, ibctesting.FirstChannelID, types.Channel{})},
+		},
+		{
+			name:   "matches port with full name as prefix",
+			prefix: transfertypes.TypeMsgTransfer,
+			allChannels: []types.IdentifiedChannel{
+				types.NewIdentifiedChannel(transfertypes.TypeMsgTransfer, "channel-0", types.Channel{}),
+				types.NewIdentifiedChannel("different-portid", "channel-1", types.Channel{}),
```

Comment:
looks like these get reused, should we specify them as local variables "transferChannel" "differentPortIDChannel"?